drivers/azure/azureutil/azureutil.go

```diff
@@ -429,9 +429,13 @@ func (a AzureClient) DeleteVirtualMachineIfExists(resourceGroup, name string) er
 	}
 
 	// Remove disk
-	if vmRef.VirtualMachineProperties != nil {
-		vhdURL := to.String(vmRef.VirtualMachineProperties.StorageProfile.OsDisk.Vhd.URI)
-		return a.removeOSDiskBlob(resourceGroup, name, vhdURL)
+	if vmProperties := vmRef.VirtualMachineProperties; vmProperties!= nil {
+		// TODO: remove unattached managed disk
```

Comment:
Does this leave the disk stranded in azure and requires manual intervention? Might log it for the user